ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java

```diff
@@ -505,24 +515,27 @@ private void executeDeleteDatasetVersionStatement(PreparedStatement statement, i
    * @param containerId the id of the container.
    * @param datasetName the name of the dataset.
    * @param version the version value of the dataset.
-   * @param expirationTimeMs the expiration time in milliseconds.
+   * @param timeToLiveInSeconds The dataset version level ttl.
+   * @param creationTimeInMs the creation time of the dataset version.
    * @param dataset the {@link Dataset} including the metadata.
-   * @return
+   * @param datasetVersionTtlEnabled set to true if dataset version ttl want to override the dataset level default ttl.
+   * @return the updated expiration time.
    * @throws SQLException
    * @throws AccountServiceException
    */
-  private long executeUpdateDatasetVersionIfExpiredSqlStatement(PreparedStatement statement,
-      int accountId, int containerId, String datasetName, long version, long expirationTimeMs, Dataset dataset)
-      throws SQLException, AccountServiceException {
+  private long executeUpdateDatasetVersionIfExpiredSqlStatement(PreparedStatement statement, int accountId,
```

Comment:
In my opinion, the statement can be created inside this function, not passed in as parameter.
Because the PreparedStatement has to match with the code below. The caller must be aware of the statement it have to pass in.

Also creating the statement inside the function is easy for reading. Reader can match the order of ? in the statement with the code set the parameters.

Reply:
I just follow the previous format based on other request. I feel like it's more clear to read the mysql statement in a common place. And most of the requests are simple and by following the order it's easy to read. My personal opinion is to keep the format consistent since both account and named blob sql statement follow the same format.

Reply:
I don't mean that we need move the insertDatasetSql definition. 
I don't see the benefit that create the PreparedStatement outside and pass in to the function as below.

        PreparedStatement updateDatasetSqlIfExpiredStatement =
            dataAccessor.getPreparedStatement(updateDatasetVersionIfExpiredSql, true);
        long newExpirationTimeMs =
              executeUpdateDatasetVersionIfExpiredSqlStatement(updateDatasetSqlIfExpiredStatement, accountId,
                  containerId, datasetName, versionNumber, expirationTimeMs, dataset);

Actually many of the other functions as below generates the PreparedStatement inside the function when the set is altogether. 
- getNewAccounts
- getContainers
- getNewContainers
- getContainerById


For some batch functions, the reason it wants to pass in the statement is it tries to share the code forinsertAccountStatement/updateAccountStatement/insertContainerStatement/updateContainerStatement

Reply:
Oh, I think the reason that those function has the statement and execution together is due to it's pretty simple. But my case is more complicated so I created the sub method to do the execution in order to make the function easy to read. And by checking the execute* method name and java doc, you will know which statement I'm execute.